es6-shim.js

```diff
@@ -285,6 +292,11 @@
       return typeof x === 'function' && _toString(x) === '[object Function]';
     },
 
+    IsConstructor: function (x) {
+      // We can't tell callables from constructors in ES5
+      return ES.IsCallable(x);
```

Comment:
Yeah, my motivation for breaking it out is just (a) to mirror the spec more closely so you can read some of our implementations and match them step-by-step to the spec, and (b) to provide a central place to put special cases for IsConstructor (like arrow functions, or whatever) when we find a good reason to do so (for compatibility in some edge case).  But it doesn't seem worth while to start cluttering up IsConstructor yet until we have the use case for it.